addons/l10n_uy/models/res_partner.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import re
+
+from odoo import api, models, _
+
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.constrains('vat', 'l10n_latam_identification_type_id')
+    def check_vat(self):
+        # EXTEND account/base_vat
+        """ Add validation of UY document types CI and NIE """
+        ci_nie_types = self.filtered(
+            lambda x: x.l10n_latam_identification_type_id.l10n_uy_dgi_code in ['1', '3']
+            and x.l10n_latam_identification_type_id.country_id.code == 'UY' and x.vat)
```

Comment:
- since it's a partner, `p` may be a better temp variable name
- tuple is more memory friendly than list. It's better to use that if we don't need the list functionality
```suggestion
        ci_nie_types = self.filtered(
            lambda p: p.l10n_latam_identification_type_id.l10n_uy_dgi_code in ('1', '3')
            and p.l10n_latam_identification_type_id.country_id.code == 'UY' and p.vat)